### PR TITLE
Use MariaDB Docker health check

### PR DESCRIPTION
**Pull request by szepeviktor**
They stole my idea!
Now it is _official_.
https://github.com/MariaDB/mariadb-docker/blob/fbb2d9a74295909ac8c50c7959c956aeb28cd150/10.5/healthcheck.sh#L50-L87

Syntax
https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck
https://docs.docker.com/engine/reference/builder/#healthcheck

### CODE REVIEW

**Comment by szepeviktor** [comment]
⚠️ I have zero clue what I am doing.

**Comment by szepeviktor** [comment]
All green in the above linked PR 🍏

**Comment by szepeviktor** [comment]
This PR is so good it gets [**copied**](https://github.com/staabm/phpstan-dba/pull/613).

**Comment by johnbillion** [comment]
Does the mysql image have a similar healthcheck? I am planning to add testing that runs on both MariaDB and MySQL.

**Comment by szepeviktor** [comment]
> Does the mysql image have a similar healthcheck?

Yes! Although barely _similar_.

```yaml
test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
```
You could add healthcheck to the matrix.

File: tests/docker-compose.yml [hunk 1]
```diff
@@ -80,8 +85,6 @@ services:
       - server
     ports:
       - 4444
-    extra_hosts:
-      - host.docker.internal:host-gateway
```
Comment (szepeviktor) [inline tests/docker-compose.yml]: Missed in https://github.com/johnbillion/plugin-infrastructure/commit/3c1582bc549d8138eeb7195ab9bd01ac1b33d923

**Comment by johnbillion** [comment]
See also https://github.com/WordPress/wordpress-develop/pull/5065/files

File: tests/docker-compose.yml (hunk 1 above)
Comment (johnbillion) [reply tests/docker-compose.yml]: This is needed for the Chrome container but not the others

Comment (szepeviktor) [reply tests/docker-compose.yml]: Reverted.

**Comment by johnbillion** [comment]
@szepeviktor Boom, we've now got automatic testing of PRs to this repo for all four of the plugins that use this library. See the links above.

If the PR is from a fork (such as this one) then the workflow needs to be manually run otherwise it would allow a PR to run an arbitrary fork of plugin-infrastructure against my plugin repos.

**Comment by szepeviktor** [comment]
All green 🍏 

Is it ready to merge?

**Comment by johnbillion** [comment]
@szepeviktor All my nightly tests started failing :( https://github.com/johnbillion/wp-crontrol/actions/workflows/nightly-tests.yml

> dependency failed to start: container wp-crontrol-database is unhealthy

**Comment by szepeviktor** [comment]
@johnbillion I've answered in #36

**Pull request merged**